actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -257,7 +312,7 @@ def to_h
     #   params.to_unsafe_h
     #   # => {"name"=>"Senjougahara Hitagi", "oddity" => "Heavy stone crab"}
     def to_unsafe_h
-      convert_parameters_to_hashes(@parameters, :to_unsafe_h)
+      permit!.to_h
```

Comment:
If it's just a simple dup that seems fine.. if it would need to be a deep dup, I think it's probably worth the effort of the more complicated version.